components/DarkModeToggle.tsx

```diff
@@ -0,0 +1,86 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+
+// SVG Icons as components
+const SunIcon = ({ className = 'w-5 h-5' }) => (
+  <svg
+    className={className}
+    fill='none'
+    stroke='white'
+    stroke-width='2'
+    viewBox='0 0 24 24'
+    xmlns='http://www.w3.org/2000/svg'
+  >
+    <circle cx='12' cy='12' r='5' />
+    <path d='M12 1v2M12 21v2M4.22 4.22l1.42 1.42M18.36 18.36l1.42 1.42M1 12h2M21 12h2M4.22 19.78l1.42-1.42M18.36 5.64l1.42-1.42' />
+  </svg>
+);
+
+const MoonIcon = ({ className = 'w-5 h-5' }) => (
+  <svg
+    className={`stroke-current text-opacity-85 text-zinc-600 ${className}`}
+    fill='none'
+    stroke-width='2'
```

Comment:
In JSX, all SVG attribute names must be written in camelCase, not kebab-case (dash-separated), because JSX follows JavaScript object property naming conventions.

Reply:
good catch, will change